argus/debug/listener.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2014 Cloudbase Solutions Srl
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from multiprocessing.connection import Listener
+
+class BasicListener():
+    """Listener implementation in case the debuger is the BasicDebuger
+    """
+    _PAUSE = 'pause'
+    _RESUME = 'resume'
+
+    def __init__(self, config):
+        """Initialize and get needed config values to avoid importing util.
+        """
+        self._ip = config.ip
+        self._port = config.port
+        self._authkey = config.authkey
+
+    def _new_listener(self):
+        """Returns a Listener according to values given in config.
+        """
+        return Listener((self._ip, self._port), authkey = self._authkey)
+
+    def start(self):
+        """Starts listening for connections and treat client responses.
+        """
+        listener = self._new_listener()
+        while True:
+            #This will wait until a client connects and the connection will
+            #persist until the client closes it.
+            conn = listener.accept()
+            typec = conn.recv_bytes()
+            msg = conn.recv_bytes()
+            if typec == self._PAUSE:
+                print(msg)
```

Comment:
I can't use log for this. You just commnet what you see but you don't know how it actually works. This happens in another window and that text is printed there for me to know what to do next and not to keep some logs.